sql/analyzer/resolve_columns.go

```diff
@@ -924,6 +948,13 @@ func resolveColumnExpression(a *Analyzer, n sql.Node, e column, columns map[tabl
 
 			// This means the expression is either a non-existent column or an alias defined in the same projection.
 			// Check for the latter first.
+			// NOTE: For GroupBy nodes, the projected expressions and grouping expressions are both returned from
```

Comment:
In the future, maybe scopes could carry passthrough and projection columns. Passthrough = selected as a return column from scope source, projection being an alias.

Reply:
That would be very cool. I started playing with something sorta similar yesterday – tracking which visible references were projected, versus which were just available from visible tables. To get `having`'s visibility rule's more correct we need more detail about that. Tracking that as part of the scope would be cool.